atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/TableElement.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.api.watch;
+
+import com.google.common.collect.Range;
+import com.palantir.lock.LockDescriptor;
+
+public interface TableElement {
```

Comment:
this doesn't seem like the right name for this type - it's a really generic type name to describe really specific behaviour. I don't know quite what the right name is, but how about LockRange?

Reply:
also - is this expected to be json serializable?

Reply:
also - I've found the Guava Range types to cause unnecessary pain, because once you've created one you can't actually back out what the range is describing.

It makes sense as an implementation detail of timelock, but feels weirder as something to slot in the Atlas API - it's like, we probably don't want anyone who's not us using this implementation - and it's not as though people can define their own ranges.

For that reason, maybe worth creating a visitor in this package and converting into Ranges within timelock, rather than in the atlasdb-api package?